### PR TITLE
feat(swap): get frontend fee from backend

### PR DESCRIPTION
**Pull request by michalszorad**
Resolves https://emurgo.atlassian.net/browse/YOMO-887

### CODE REVIEW

File: packages/swap/src/adapters/openswap-api/api.ts [hunk 1]
```diff
@@ -108,6 +108,10 @@ export const swapApiMaker = (
       .then((response) => response.price)
   }
 
+  const getFrontendFees: Swap.Api['getFrontendFees'] = async () => {
```
Comment (michalszorad) [inline packages/swap/src/adapters/openswap-api/api.ts:111]: Should this be removed as well and moved to yoroi mobile?

File: packages/swap/src/helpers/orders/factories/makeOrderCalculations.ts [hunk 2]
```diff
@@ -31,10 +32,15 @@ export const makeOrderCalculations = ({
   slippage: number
   primaryTokenId: Balance.TokenInfo['id']
   side?: 'buy' | 'sell'
+  discountTiers: ReadonlyArray<Swap.DiscountTier> | undefined
 }>): Array<SwapOrderCalculation> => {
   const isLimit = orderType === 'limit'
   const maybeLimitPrice = isLimit ? limitPrice : undefined
 
+  if (!discountTiers || discountTiers.length === 0) {
+    throw new Error('discountTiers is undefined')
```
Comment (michalszorad) [inline packages/swap/src/helpers/orders/factories/makeOrderCalculations.ts]: What is the plan to respond from the backend? Empty discount tiers?

File: packages/openswap/src/api.ts [hunk 3]
```diff
@@ -103,6 +105,10 @@ export class OpenSwapApi {
 
     return tokens
   }
+
+  public async getFrontendFees(): Promise<Swap.FrontendFee> {
```
Comment (michalszorad) [inline packages/openswap/src/api.ts:109]: Should this be removed and moved to yoroi mobile?

File: apps/wallet-mobile/src/features/Swap/SwapNavigator.tsx [hunk 4]
```diff
@@ -16,14 +16,22 @@ const Tab = createMaterialTopTabNavigator<SwapTabRoutes>()
...
   const {refetch} = useSwapTokensByPairToken('', {suspense: false, enabled: false})
 
   React.useEffect(() => {
     refetch()
   }, [refetch])
 
+  React.useEffect(() => {
+    discountTiersChanged(tiers)
+  }, [discountTiersChanged, tiers])
+
```
Comment (michalszorad) [inline apps/wallet-mobile/src/features/Swap/SwapNavigator.tsx:34]: Should we block the UI somehow when fees are not fetched yet? What if the request fails?

File: packages/openswap/src/fee.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import {asQuantity} from '@yoroi/swap/src/utils/asQuantity'
+import {Quantities} from '@yoroi/swap/src/utils/quantities'
+import {ApiDeps, GetFrontedFeeResponse} from './types'
+import {Swap} from '@yoroi/types'
+
+export const fetchFrontendFee = async (
+  _deps: ApiDeps,
+): Promise<GetFrontedFeeResponse> => {
+  return {
+    muesliswap: {
```
Comment (michalszorad) [inline packages/openswap/src/fee.ts:10]: What is the endpoint?

File: packages/openswap/src/fee.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    secondaryTokenBalanceThreshold: Quantities.zero,
+    variableFeeMultiplier: 0.0005,
+    variableFeeVisual: 0.05,
+    fixedFee: asQuantity(1_000_000),
+  },
+  // VALUE ADA 0-99, FFEE = 0%
+]
+
+const milkTokenId = {
+  mainnet: '8a1cfae21368b8bebbbed9800fec304e95cce39a2a57dc35e2e3ebaa.4d494c4b',
```
Comment (michalszorad) [inline packages/openswap/src/fee.ts:46]: Where is this being compared?

File: apps/wallet-mobile/src/features/Swap/SwapNavigator.tsx [hunk 7]
```diff
@@ -16,14 +16,22 @@ const Tab = createMaterialTopTabNavigator<SwapTabRoutes>()
...
 
+  const {data: tiers = []} = useFrontendFees()
+
   const {refetch} = useSwapTokensByPairToken('', {suspense: false, enabled: false})
 
   React.useEffect(() => {
     refetch()
   }, [refetch])
 
+  React.useEffect(() => {
```
Comment (michalszorad) [inline apps/wallet-mobile/src/features/Swap/SwapNavigator.tsx:31]: Should we also block the UI if fee > 0 as this case is not properly implemented yet? Or how should the UI behave if the fee is > 0?

File: packages/openswap/src/fee.ts (hunk 5 above)
Comment (michalszorad) [reply packages/openswap/src/fee.ts:10]: We're using these mocked data in the meantime

**Comment by stackchain** [comment]
I think we can close in favour of #2807

**Comment by michalszorad** [comment]
@stackchain sounds good to me

**Pull request closed without merging**